xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/displayers/actions/BaseAction.vue

```diff
@@ -0,0 +1,78 @@
+<!--
+  See the NOTICE file distributed with this work for additional
+  information regarding copyright ownership.
+  This is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as
+  published by the Free Software Foundation; either version 2.1 of
+  the License, or (at your option) any later version.
+  This software is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+  You should have received a copy of the GNU Lesser General Public
+  License along with this software; if not, write to the Free
+  Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<template>
+  <span class="livedata-base-action btn"
+        @click="trigger"
+        @keypress.self.enter="trigger"
+        :title="$t(titleTranslationKey)"
+        tabindex="0">
+    <XWikiIcon :icon-descriptor="iconDescriptor"/>
+    <template v-if="labelTranslationKey"> {{ $t(labelTranslationKey) }}</template>
+  </span>
+</template>
+
+<script>
+import XWikiIcon from "../../utilities/XWikiIcon";
+
+export default {
+
+  name: "BaseAction",
+
+  components: {
+    XWikiIcon,
+  },
+
+  props: {
+    titleTranslationKey: {
+      type: String,
+      required: true
+    },
+    closePopover: {
```

Comment:
Having to pass the `closePopover` method in every action component all the way down to this base action component seems quite burdensome to me. Maybe using a [provide / inject](https://vuejs.org/v2/api/#provide-inject) syntax would help to make this less cumbersome, it might be worth to test.

Reply:
Well, all the way down it two levels deep so it is quite straightforward, also in the future the actions could have different behaviors on close.